### PR TITLE
Support for AVM v10

### PR DESCRIPTION
**Pull request by jasonpaulos**
- [x] Enable v10 programs
- [x] `box_splice` and `box_resize` opcodes
- [x] Global fields `AssetCreateMinBalance`, `AssetOptInMinBalance`, and `GenesisHash`
- [x] Elliptic curve opcodes

### CODE REVIEW

File: pyteal/compiler/compiler.py [hunk 1]
```diff
@@ -306,6 +306,7 @@ def __init__(
         *,
         version: int = DEFAULT_PROGRAM_VERSION,
         assemble_constants: bool = False,
+        assembly_type_track: bool = True,
```
Comment (jasonpaulos) [inline pyteal/compiler/compiler.py:309]: Also added the option to include `#pragma typetrack false`. This was enabled for integration tests briefly due to the assembler type error, but the latest docker image has that fix included, so it's no longer needed.

Still, I though it'd be useful to include this.

**Review by jannotti — APPROVED** [review]
I believe I figured out that the answer to my question about "EllipticCurve" is that it's teh way to name the argument to the various opcodes?

File: docs/accessing_transaction_field.rst [hunk 2]
```diff
@@ -237,4 +237,10 @@ Operator                                    Type                    Min Program
 :any:`Global.creator_address()`             :code:`TealType.bytes`  3                    32 byte address of the creator of the current application
 :any:`Global.current_application_address()` :code:`TealType.bytes`  5                    32 byte address of the current application controlled account
 :any:`Global.group_id()`                    :code:`TealType.bytes`  5                    32 byte ID of the current transaction group
-=========================================== ======================= ==================== =============================================================
+:any:`Global.opcode_budget()`               :code:`TealType.uint64` 6                    The remaining cost that can be spent by opcodes in this program
+:any:`Global.caller_app_id()`               :code:`TealType.uint64` 6                    The ID of the application that called the current application, or zero. Application mode only
+:any:`Global.caller_app_address()`          :code:`TealType.bytes`  6                    32 byte address of the application that called the current application, or the zero address. Application mode only.
```
[the remark this answers is not in the document]
Comment (jasonpaulos) [reply docs/accessing_transaction_field.rst:242]: Yes, this file was outdated

File: pyteal/ast/__init__.py [hunk 3]
```diff
@@ -252,6 +265,13 @@
     "EcdsaVerify",
     "Ed25519Verify_Bare",
     "Ed25519Verify",
+    "EllipticCurve",
```
[the remark this answers is not in the document]
Comment (jasonpaulos) [reply pyteal/ast/__init__.py:268]: This is an enum that's required to be passed as the first arg to the new `Ec` operations. For example,

```python
EcAdd(EllipticCurve.BN254g1, a, b)
```

**Pull request merged**